### ISSUE TITLE
[1.3.0] updateFull is missing caller for direct dependencies

### ISSUE THREAD

**Issue by jrudolph** [sbt/sbt#4706]
## steps

git clone akka/akka
sbt "show akka-stream/updateFull" > update.txt

## problem

```
[info] 	compile:
[info] 	com.typesafe:config
[info] 		- 1.3.3
[info] 			evicted: false
[info] 			homepage: https://github.com/lightbend/config
[info] 			configurations: test, optional, compile, default, runtime
[info] 			callers: com.typesafe:ssl-config-core_2.12:0.3.7, com.typesafe.akka:akka-actor_2.12:2.5-SNAPSHOT
[info] 
[info] 	org.scala-lang:scala-library
[info] 		- 2.12.8
[info] 			evicted: false
[info] 			homepage: http://www.scala-lang.org/
[info] 			extraAttributes: Map(info.apiURL -> http://www.scala-lang.org/api/2.12.8/)
[info] 			configurations: test, optional, compile, default, runtime
[info] 			callers: com.typesafe.akka:akka-protobuf_2.12:2.5-SNAPSHOT, org.scala-lang.modules:scala-parser-combinators_2.12:1.1.1, com.typesafe:ssl-config-core_2.12:0.3.7, org.scala-lang.modules:scala-java8-compat_2.12:0.8.0, com.typesafe.akka:akka-actor_2.12:2.5-SNAPSHOT
[info] 
[info] 	org.scala-lang.modules:scala-parser-combinators_2.12
[info] 		- 1.1.1
[info] 			evicted: false
[info] 			homepage: http://www.scala-lang.org/
[info] 			configurations: test, optional, compile, default, runtime
[info] 			callers: com.typesafe:ssl-config-core_2.12:0.3.7
[info] 
[info] 	org.reactivestreams:reactive-streams
[info] 		- 1.0.2
[info] 			evicted: false
[info] 			homepage: http://www.reactive-streams.org/
[info] 			configurations: test, optional, compile, default, runtime
[info] 
[info] 	org.scala-lang.modules:scala-java8-compat_2.12
[info] 		- 0.8.0
[info] 			evicted: false
[info] 			homepage: http://www.scala-lang.org/
[info] 			configurations: test, optional, compile, default, runtime
[info] 			callers: com.typesafe.akka:akka-actor_2.12:2.5-SNAPSHOT
[info] 
[info] 	com.typesafe:ssl-config-core_2.12
[info] 		- 0.3.7
[info] 			evicted: false
[info] 			homepage: https://github.com/lightbend/ssl-config
[info] 			configurations: test, optional, compile, default, runtime
[info] 
```

caller info is missing for org.reactivestreams:reactive-streams and com.typesafe:ssl-config-core_2.12.

## expectation

All modules have a caller.

## original report by jrudolph

That's a regression compared to previous versions that prevents sbt-dependency-graph from working correctly.

See also
- https://github.com/jrudolph/sbt-dependency-graph/issues/178
- https://github.com/coursier/coursier/issues/1170

**Comment by eed3si9n** [comment]
Sorry. In https://github.com/sbt/sbt/pull/4606 I trimmed update report with the information.
You might need to define a dynamic task to call `updateFull` for sbt 1.3.0 and up.

**Comment by jrudolph** [comment]
Thanks for the info. However, I'll need some solution that will work across sbt 1.x versions. How would that work? It seems `updateFull` is only available from 1.3.x?

**Comment by jrudolph** [comment]
In fact, as I need to overwrite `withMissingOk` for a new task, I already used `Classpaths.updateTask` which seems to be the implementation of `updateFull` so it should still work. However, the `details` seem to be empty and also caller information is incomplete.

Another big problem is that eviction is now working differently than before. How was coursier support tested if that most basic of features has not been tested for regressions?

For me that looks like coursier support is not yet stable enough to be the default.

As a test case, you can use the example from sbt-dependency-graph:

```scala
version := "0.1.0-SNAPSHOT"

scalaVersion := "2.9.1"

resolvers += "typesafe maven" at "https://repo.typesafe.com/typesafe/maven-releases/"

libraryDependencies ++= Seq(
  "com.codahale" % "jerkson_2.9.1" % "0.5.0",
  "org.codehaus.jackson" % "jackson-mapper-asl" % "1.9.10" // as another version of asl
)
```

Output of `show update` in sbt 1.2.8:

```
[info] 	compile:
[info] 	com.codahale:jerkson_2.9.1
[info] 		- 0.5.0
[info] 			status: release
[info] 			publicationDate: 2011-10-08T02:47:46+02:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			isDefault: false
[info] 			configurations: default, compile, runtime, default(compile), master
[info] 			licenses: (The MIT License,Some(http://codahale.com/mit.txt))
[info] 			callers: default:test_2.9.1:0.1.0-SNAPSHOT
[info]
[info] 	org.codehaus.jackson:jackson-mapper-asl
[info] 		- 1.9.11
[info] 			status: release
[info] 			publicationDate: 2012-11-06T17:27:45+01:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			homepage: http://jackson.codehaus.org
[info] 			isDefault: false
[info] 			configurations: master(*), default, compile, runtime(*), runtime, compile(*), master, default(compile)
[info] 			licenses: (The Apache Software License, Version 2.0,Some(http://www.apache.org/licenses/LICENSE-2.0.txt))
[info] 			callers: default:test_2.9.1:0.1.0-SNAPSHOT, com.codahale:jerkson_2.9.1:0.5.0
[info] 		- 1.9.10
[info] 			evicted: true
[info] 			evictedData: latest-revision
[info] 			configurations: default, compile, runtime, default(compile), master
[info] 			callers: default:test_2.9.1:0.1.0-SNAPSHOT
[info]
[info] 	org.codehaus.jackson:jackson-core-asl
[info] 		- 1.9.11
[info] 			status: release
[info] 			publicationDate: 2012-11-06T17:27:13+01:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			homepage: http://jackson.codehaus.org
[info] 			isDefault: false
[info] 			configurations: master(*), compile, runtime(*), runtime, compile(*), master
[info] 			licenses: (The Apache Software License, Version 2.0,Some(http://www.apache.org/licenses/LICENSE-2.0.txt))
[info] 			callers: org.codehaus.jackson:jackson-mapper-asl:1.9.11, com.codahale:jerkson_2.9.1:0.5.0
[info]
[info] 	org.scala-lang:scala-library
[info] 		- 2.9.1
[info] 			status: release
[info] 			publicationDate: 2011-08-31T10:39:53+02:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			homepage: http://www.scala-lang.org/
[info] 			isDefault: false
[info] 			configurations: default, master(*), compile, runtime(*), runtime, default(compile), compile(*), master
[info] 			licenses: (BSD-like,Some(http://www.scala-lang.org/downloads/license.html))
[info] 			callers: default:test_2.9.1:0.1.0-SNAPSHOT, com.codahale:jerkson_2.9.1:0.5.0
[info]
[info] 	runtime:
[info] 	com.codahale:jerkson_2.9.1
[info] 		- 0.5.0
[info] 			status: release
[info] 			publicationDate: 2011-10-08T02:47:46+02:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			isDefault: false
[info] 			configurations: default, compile, runtime, default(compile), master
[info] 			licenses: (The MIT License,Some(http://codahale.com/mit.txt))
[info] 			callers: default:test_2.9.1:0.1.0-SNAPSHOT
[info]
[info] 	org.codehaus.jackson:jackson-mapper-asl
[info] 		- 1.9.11
[info] 			status: release
[info] 			publicationDate: 2012-11-06T17:27:45+01:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			homepage: http://jackson.codehaus.org
[info] 			isDefault: false
[info] 			configurations: master(*), compile, runtime(*), runtime, compile(*), master, default(compile)
[info] 			licenses: (The Apache Software License, Version 2.0,Some(http://www.apache.org/licenses/LICENSE-2.0.txt))
[info] 			callers: default:test_2.9.1:0.1.0-SNAPSHOT, com.codahale:jerkson_2.9.1:0.5.0
[info] 		- 1.9.10
[info] 			evicted: true
[info] 			evictedData: latest-revision
[info] 			configurations: default, compile, runtime, default(compile), master
[info] 			callers: default:test_2.9.1:0.1.0-SNAPSHOT
[info]
[info] 	org.codehaus.jackson:jackson-core-asl
[info] 		- 1.9.11
[info] 			status: release
[info] 			publicationDate: 2012-11-06T17:27:13+01:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			homepage: http://jackson.codehaus.org
[info] 			isDefault: false
[info] 			configurations: master(*), compile, runtime(*), runtime, compile(*), master
[info] 			licenses: (The Apache Software License, Version 2.0,Some(http://www.apache.org/licenses/LICENSE-2.0.txt))
[info] 			callers: org.codehaus.jackson:jackson-mapper-asl:1.9.11, com.codahale:jerkson_2.9.1:0.5.0
[info]
[info] 	org.scala-lang:scala-library
[info] 		- 2.9.1
[info] 			status: release
[info] 			publicationDate: 2011-08-31T10:39:53+02:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			homepage: http://www.scala-lang.org/
[info] 			isDefault: false
[info] 			configurations: default, master(*), compile, runtime(*), runtime, default(compile), compile(*), master
[info] 			licenses: (BSD-like,Some(http://www.scala-lang.org/downloads/license.html))
[info] 			callers: default:test_2.9.1:0.1.0-SNAPSHOT, com.codahale:jerkson_2.9.1:0.5.0
[info]
[info] 	test:
[info] 	com.codahale:jerkson_2.9.1
[info] 		- 0.5.0
[info] 			status: release
[info] 			publicationDate: 2011-10-08T02:47:46+02:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			isDefault: false
[info] 			configurations: default, compile, runtime, default(compile), master
[info] 			licenses: (The MIT License,Some(http://codahale.com/mit.txt))
[info] 			callers: default:test_2.9.1:0.1.0-SNAPSHOT
[info]
[info] 	org.codehaus.jackson:jackson-mapper-asl
[info] 		- 1.9.11
[info] 			status: release
[info] 			publicationDate: 2012-11-06T17:27:45+01:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			homepage: http://jackson.codehaus.org
[info] 			isDefault: false
[info] 			configurations: master(*), compile, runtime(*), runtime, compile(*), master, default(compile)
[info] 			licenses: (The Apache Software License, Version 2.0,Some(http://www.apache.org/licenses/LICENSE-2.0.txt))
[info] 			callers: default:test_2.9.1:0.1.0-SNAPSHOT, com.codahale:jerkson_2.9.1:0.5.0
[info] 		- 1.9.10
[info] 			evicted: true
[info] 			evictedData: latest-revision
[info] 			configurations: default, compile, runtime, default(compile), master
[info] 			callers: default:test_2.9.1:0.1.0-SNAPSHOT
[info]
[info] 	org.codehaus.jackson:jackson-core-asl
[info] 		- 1.9.11
[info] 			status: release
[info] 			publicationDate: 2012-11-06T17:27:13+01:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			homepage: http://jackson.codehaus.org
[info] 			isDefault: false
[info] 			configurations: master(*), compile, runtime(*), runtime, compile(*), master
[info] 			licenses: (The Apache Software License, Version 2.0,Some(http://www.apache.org/licenses/LICENSE-2.0.txt))
[info] 			callers: org.codehaus.jackson:jackson-mapper-asl:1.9.11, com.codahale:jerkson_2.9.1:0.5.0
[info]
[info] 	org.scala-lang:scala-library
[info] 		- 2.9.1
[info] 			status: release
[info] 			publicationDate: 2011-08-31T10:39:53+02:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			homepage: http://www.scala-lang.org/
[info] 			isDefault: false
[info] 			configurations: default, master(*), compile, runtime(*), runtime, default(compile), compile(*), master
[info] 			licenses: (BSD-like,Some(http://www.scala-lang.org/downloads/license.html))
[info] 			callers: default:test_2.9.1:0.1.0-SNAPSHOT, com.codahale:jerkson_2.9.1:0.5.0
[info]
[info] 	provided:
[info] 	optional:
[info] 	compile-internal:
[info] 	com.codahale:jerkson_2.9.1
[info] 		- 0.5.0
[info] 			status: release
[info] 			publicationDate: 2011-10-08T02:47:46+02:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			isDefault: false
[info] 			configurations: default, compile, runtime, default(compile), master
[info] 			licenses: (The MIT License,Some(http://codahale.com/mit.txt))
[info] 			callers: default:test_2.9.1:0.1.0-SNAPSHOT
[info]
[info] 	org.codehaus.jackson:jackson-mapper-asl
[info] 		- 1.9.11
[info] 			status: release
[info] 			publicationDate: 2012-11-06T17:27:45+01:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			homepage: http://jackson.codehaus.org
[info] 			isDefault: false
[info] 			configurations: master(*), compile, runtime(*), runtime, compile(*), master, default(compile)
[info] 			licenses: (The Apache Software License, Version 2.0,Some(http://www.apache.org/licenses/LICENSE-2.0.txt))
[info] 			callers: default:test_2.9.1:0.1.0-SNAPSHOT, com.codahale:jerkson_2.9.1:0.5.0
[info] 		- 1.9.10
[info] 			evicted: true
[info] 			evictedData: latest-revision
[info] 			configurations: default, compile, runtime, default(compile), master
[info] 			callers: default:test_2.9.1:0.1.0-SNAPSHOT
[info]
[info] 	org.codehaus.jackson:jackson-core-asl
[info] 		- 1.9.11
[info] 			status: release
[info] 			publicationDate: 2012-11-06T17:27:13+01:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			homepage: http://jackson.codehaus.org
[info] 			isDefault: false
[info] 			configurations: master(*), compile, runtime(*), runtime, compile(*), master
[info] 			licenses: (The Apache Software License, Version 2.0,Some(http://www.apache.org/licenses/LICENSE-2.0.txt))
[info] 			callers: org.codehaus.jackson:jackson-mapper-asl:1.9.11, com.codahale:jerkson_2.9.1:0.5.0
[info]
[info] 	org.scala-lang:scala-library
[info] 		- 2.9.1
[info] 			status: release
[info] 			publicationDate: 2011-08-31T10:39:53+02:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			homepage: http://www.scala-lang.org/
[info] 			isDefault: false
[info] 			configurations: default, master(*), compile, runtime(*), runtime, default(compile), compile(*), master
[info] 			licenses: (BSD-like,Some(http://www.scala-lang.org/downloads/license.html))
[info] 			callers: default:test_2.9.1:0.1.0-SNAPSHOT, com.codahale:jerkson_2.9.1:0.5.0
[info]
[info] 	runtime-internal:
[info] 	com.codahale:jerkson_2.9.1
[info] 		- 0.5.0
[info] 			status: release
[info] 			publicationDate: 2011-10-08T02:47:46+02:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			isDefault: false
[info] 			configurations: default, compile, runtime, default(compile), master
[info] 			licenses: (The MIT License,Some(http://codahale.com/mit.txt))
[info] 			callers: default:test_2.9.1:0.1.0-SNAPSHOT
[info]
[info] 	org.codehaus.jackson:jackson-mapper-asl
[info] 		- 1.9.11
[info] 			status: release
[info] 			publicationDate: 2012-11-06T17:27:45+01:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			homepage: http://jackson.codehaus.org
[info] 			isDefault: false
[info] 			configurations: master(*), compile, runtime(*), runtime, compile(*), master, default(compile)
[info] 			licenses: (The Apache Software License, Version 2.0,Some(http://www.apache.org/licenses/LICENSE-2.0.txt))
[info] 			callers: default:test_2.9.1:0.1.0-SNAPSHOT, com.codahale:jerkson_2.9.1:0.5.0
[info] 		- 1.9.10
[info] 			evicted: true
[info] 			evictedData: latest-revision
[info] 			configurations: default, compile, runtime, default(compile), master
[info] 			callers: default:test_2.9.1:0.1.0-SNAPSHOT
[info]
[info] 	org.codehaus.jackson:jackson-core-asl
[info] 		- 1.9.11
[info] 			status: release
[info] 			publicationDate: 2012-11-06T17:27:13+01:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			homepage: http://jackson.codehaus.org
[info] 			isDefault: false
[info] 			configurations: master(*), compile, runtime(*), runtime, compile(*), master
[info] 			licenses: (The Apache Software License, Version 2.0,Some(http://www.apache.org/licenses/LICENSE-2.0.txt))
[info] 			callers: org.codehaus.jackson:jackson-mapper-asl:1.9.11, com.codahale:jerkson_2.9.1:0.5.0
[info]
[info] 	org.scala-lang:scala-library
[info] 		- 2.9.1
[info] 			status: release
[info] 			publicationDate: 2011-08-31T10:39:53+02:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			homepage: http://www.scala-lang.org/
[info] 			isDefault: false
[info] 			configurations: default, master(*), compile, runtime(*), runtime, default(compile), compile(*), master
[info] 			licenses: (BSD-like,Some(http://www.scala-lang.org/downloads/license.html))
[info] 			callers: default:test_2.9.1:0.1.0-SNAPSHOT, com.codahale:jerkson_2.9.1:0.5.0
[info]
[info] 	test-internal:
[info] 	com.codahale:jerkson_2.9.1
[info] 		- 0.5.0
[info] 			status: release
[info] 			publicationDate: 2011-10-08T02:47:46+02:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			isDefault: false
[info] 			configurations: default, compile, runtime, default(compile), master
[info] 			licenses: (The MIT License,Some(http://codahale.com/mit.txt))
[info] 			callers: default:test_2.9.1:0.1.0-SNAPSHOT
[info]
[info] 	org.codehaus.jackson:jackson-mapper-asl
[info] 		- 1.9.11
[info] 			status: release
[info] 			publicationDate: 2012-11-06T17:27:45+01:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			homepage: http://jackson.codehaus.org
[info] 			isDefault: false
[info] 			configurations: master(*), compile, runtime(*), runtime, compile(*), master, default(compile)
[info] 			licenses: (The Apache Software License, Version 2.0,Some(http://www.apache.org/licenses/LICENSE-2.0.txt))
[info] 			callers: default:test_2.9.1:0.1.0-SNAPSHOT, com.codahale:jerkson_2.9.1:0.5.0
[info] 		- 1.9.10
[info] 			evicted: true
[info] 			evictedData: latest-revision
[info] 			configurations: default, compile, runtime, default(compile), master
[info] 			callers: default:test_2.9.1:0.1.0-SNAPSHOT
[info]
[info] 	org.codehaus.jackson:jackson-core-asl
[info] 		- 1.9.11
[info] 			status: release
[info] 			publicationDate: 2012-11-06T17:27:13+01:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			homepage: http://jackson.codehaus.org
[info] 			isDefault: false
[info] 			configurations: master(*), compile, runtime(*), runtime, compile(*), master
[info] 			licenses: (The Apache Software License, Version 2.0,Some(http://www.apache.org/licenses/LICENSE-2.0.txt))
[info] 			callers: org.codehaus.jackson:jackson-mapper-asl:1.9.11, com.codahale:jerkson_2.9.1:0.5.0
[info]
[info] 	org.scala-lang:scala-library
[info] 		- 2.9.1
[info] 			status: release
[info] 			publicationDate: 2011-08-31T10:39:53+02:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			homepage: http://www.scala-lang.org/
[info] 			isDefault: false
[info] 			configurations: default, master(*), compile, runtime(*), runtime, default(compile), compile(*), master
[info] 			licenses: (BSD-like,Some(http://www.scala-lang.org/downloads/license.html))
[info] 			callers: default:test_2.9.1:0.1.0-SNAPSHOT, com.codahale:jerkson_2.9.1:0.5.0
[info]
[info] 	plugin:
[info] 	pom:
[info] 	scala-tool:
[info] 	org.scala-lang:scala-compiler
[info] 		- 2.9.1
[info] 			status: release
[info] 			publicationDate: 2011-08-31T10:41:08+02:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			homepage: http://www.scala-lang.org/
[info] 			isDefault: false
[info] 			configurations: default, compile, runtime, optional, optional(default), master
[info] 			licenses: (BSD-like,Some(http://www.scala-lang.org/downloads/license.html))
[info] 			callers: default:test_2.9.1:0.1.0-SNAPSHOT
[info]
[info] 	org.scala-lang:jline
[info] 		- 2.9.1
[info] 			status: release
[info] 			publicationDate: 2011-08-31T10:40:46+02:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			homepage: http://www.scala-lang.org/
[info] 			isDefault: false
[info] 			configurations: master(*), compile, compile(*), master
[info] 			licenses: (BSD-like,Some(http://www.scala-lang.org/downloads/license.html)), (The BSD License,Some(http://www.opensource.org/licenses/bsd-license.php))
[info] 			callers: org.scala-lang:scala-compiler:2.9.1
[info]
[info] 	org.fusesource.jansi:jansi
[info] 		- 1.4
[info] 			status: release
[info] 			publicationDate: 2010-07-15T16:21:57+02:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			homepage: http://${forge-project-id}.fusesource.org
[info] 			isDefault: false
[info] 			configurations: master(*), compile, compile(*), master
[info] 			licenses: (The Apache Software License, Version 2.0,Some(http://www.apache.org/licenses/LICENSE-2.0.txt))
[info] 			callers: org.scala-lang:jline:2.9.1
[info]
[info] 	org.scala-lang:scala-library
[info] 		- 2.9.1
[info] 			status: release
[info] 			publicationDate: 2011-08-31T10:39:53+02:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			homepage: http://www.scala-lang.org/
[info] 			isDefault: false
[info] 			configurations: default, master(*), compile, runtime(*), runtime, optional, optional(default), compile(*), master
[info] 			licenses: (BSD-like,Some(http://www.scala-lang.org/downloads/license.html))
[info] 			callers: org.scala-lang:scala-compiler:2.9.1, default:test_2.9.1:0.1.0-SNAPSHOT
[info]
```

Output of `show updateFull` in sbt 1.3.0-RC1:

```
[info]  plugin:
[info]  pom:
[info]  test:
[info]          com.codahale:jerkson_2.9.1:0.5.0:default (artifactId=jerkson_2.9.1, project.version=0.5.0, pom.groupId=com.codahale, groupId=com.codahale, project.groupId=com.codahale, version=0.5.0, pom.version=0.5.0, project.artifactId=jerkson_2.9.1, jackson.version=[1.9.0,2.0.0), scala.version=2.9.1, pom.artifactId=jerkson_2.9.1): (Artifact(jerkson_2.9.1, jar, jar, None, Vector(), Some(https://repo.typesafe.com/typesafe/maven-releases/com/codahale/jerkson_2.9.1/0.5.0/jerkson_2.9.1-0.5.0.jar), Map(artifactId -> jerkson_2.9.1, project.version -> 0.5.0, pom.groupId -> com.codahale, groupId -> com.codahale, project.groupId -> com.codahale, version -> 0.5.0, pom.version -> 0.5.0, project.artifactId -> jerkson_2.9.1, jackson.version -> [1.9.0,2.0.0), scala.version -> 2.9.1, pom.artifactId -> jerkson_2.9.1), None),/home/johannes/.coursier/cache/v1/https/repo.typesafe.com/typesafe/maven-releases/com/codahale/jerkson_2.9.1/0.5.0/jerkson_2.9.1-0.5.0.jar)
[info]          org.scala-lang:scala-library:2.9.1:default (artifactId=scala-library, project.version=2.9.1, pom.groupId=org.scala-lang, groupId=org.scala-lang, project.groupId=org.scala-lang, version=2.9.1, pom.version=2.9.1, project.artifactId=scala-library, pom.artifactId=scala-library): (Artifact(scala-library, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/scala-lang/scala-library/2.9.1/scala-library-2.9.1.jar), Map(artifactId -> scala-library, project.version -> 2.9.1, pom.groupId -> org.scala-lang, groupId -> org.scala-lang, project.groupId -> org.scala-lang, version -> 2.9.1, pom.version -> 2.9.1, project.artifactId -> scala-library, pom.artifactId -> scala-library), None),/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.9.1/scala-library-2.9.1.jar)
[info]          org.codehaus.jackson:jackson-mapper-asl:1.9.10:default (artifactId=jackson-mapper-asl, project.version=1.9.10, pom.groupId=org.codehaus.jackson, groupId=org.codehaus.jackson, project.groupId=org.codehaus.jackson, version=1.9.10, pom.version=1.9.10, project.artifactId=jackson-mapper-asl, pom.artifactId=jackson-mapper-asl): (Artifact(jackson-mapper-asl, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/codehaus/jackson/jackson-mapper-asl/1.9.10/jackson-mapper-asl-1.9.10.jar), Map(artifactId -> jackson-mapper-asl, project.version -> 1.9.10, pom.groupId -> org.codehaus.jackson, groupId -> org.codehaus.jackson, project.groupId -> org.codehaus.jackson, version -> 1.9.10, pom.version -> 1.9.10, project.artifactId -> jackson-mapper-asl, pom.artifactId -> jackson-mapper-asl), None),/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/org/codehaus/jackson/jackson-mapper-asl/1.9.10/jackson-mapper-asl-1.9.10.jar)
[info]          org.codehaus.jackson:jackson-core-asl:1.9.10:default (artifactId=jackson-core-asl, project.version=1.9.10, pom.groupId=org.codehaus.jackson, groupId=org.codehaus.jackson, project.groupId=org.codehaus.jackson, version=1.9.10, pom.version=1.9.10, project.artifactId=jackson-core-asl, pom.artifactId=jackson-core-asl): (Artifact(jackson-core-asl, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/codehaus/jackson/jackson-core-asl/1.9.10/jackson-core-asl-1.9.10.jar), Map(artifactId -> jackson-core-asl, project.version -> 1.9.10, pom.groupId -> org.codehaus.jackson, groupId -> org.codehaus.jackson, project.groupId -> org.codehaus.jackson, version -> 1.9.10, pom.version -> 1.9.10, project.artifactId -> jackson-core-asl, pom.artifactId -> jackson-core-asl), None),/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/org/codehaus/jackson/jackson-core-asl/1.9.10/jackson-core-asl-1.9.10.jar)
[info]  provided:
[info]  compile-internal:
[info]          com.codahale:jerkson_2.9.1:0.5.0:default (artifactId=jerkson_2.9.1, project.version=0.5.0, pom.groupId=com.codahale, groupId=com.codahale, project.groupId=com.codahale, version=0.5.0, pom.version=0.5.0, project.artifactId=jerkson_2.9.1, jackson.version=[1.9.0,2.0.0), scala.version=2.9.1, pom.artifactId=jerkson_2.9.1): (Artifact(jerkson_2.9.1, jar, jar, None, Vector(), Some(https://repo.typesafe.com/typesafe/maven-releases/com/codahale/jerkson_2.9.1/0.5.0/jerkson_2.9.1-0.5.0.jar), Map(artifactId -> jerkson_2.9.1, project.version -> 0.5.0, pom.groupId -> com.codahale, groupId -> com.codahale, project.groupId -> com.codahale, version -> 0.5.0, pom.version -> 0.5.0, project.artifactId -> jerkson_2.9.1, jackson.version -> [1.9.0,2.0.0), scala.version -> 2.9.1, pom.artifactId -> jerkson_2.9.1), None),/home/johannes/.coursier/cache/v1/https/repo.typesafe.com/typesafe/maven-releases/com/codahale/jerkson_2.9.1/0.5.0/jerkson_2.9.1-0.5.0.jar)
[info]          org.scala-lang:scala-library:2.9.1:default (artifactId=scala-library, project.version=2.9.1, pom.groupId=org.scala-lang, groupId=org.scala-lang, project.groupId=org.scala-lang, version=2.9.1, pom.version=2.9.1, project.artifactId=scala-library, pom.artifactId=scala-library): (Artifact(scala-library, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/scala-lang/scala-library/2.9.1/scala-library-2.9.1.jar), Map(artifactId -> scala-library, project.version -> 2.9.1, pom.groupId -> org.scala-lang, groupId -> org.scala-lang, project.groupId -> org.scala-lang, version -> 2.9.1, pom.version -> 2.9.1, project.artifactId -> scala-library, pom.artifactId -> scala-library), None),/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.9.1/scala-library-2.9.1.jar)
[info]          org.codehaus.jackson:jackson-mapper-asl:1.9.10:default (artifactId=jackson-mapper-asl, project.version=1.9.10, pom.groupId=org.codehaus.jackson, groupId=org.codehaus.jackson, project.groupId=org.codehaus.jackson, version=1.9.10, pom.version=1.9.10, project.artifactId=jackson-mapper-asl, pom.artifactId=jackson-mapper-asl): (Artifact(jackson-mapper-asl, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/codehaus/jackson/jackson-mapper-asl/1.9.10/jackson-mapper-asl-1.9.10.jar), Map(artifactId -> jackson-mapper-asl, project.version -> 1.9.10, pom.groupId -> org.codehaus.jackson, groupId -> org.codehaus.jackson, project.groupId -> org.codehaus.jackson, version -> 1.9.10, pom.version -> 1.9.10, project.artifactId -> jackson-mapper-asl, pom.artifactId -> jackson-mapper-asl), None),/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/org/codehaus/jackson/jackson-mapper-asl/1.9.10/jackson-mapper-asl-1.9.10.jar)
[info]          org.codehaus.jackson:jackson-core-asl:1.9.10:default (artifactId=jackson-core-asl, project.version=1.9.10, pom.groupId=org.codehaus.jackson, groupId=org.codehaus.jackson, project.groupId=org.codehaus.jackson, version=1.9.10, pom.version=1.9.10, project.artifactId=jackson-core-asl, pom.artifactId=jackson-core-asl): (Artifact(jackson-core-asl, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/codehaus/jackson/jackson-core-asl/1.9.10/jackson-core-asl-1.9.10.jar), Map(artifactId -> jackson-core-asl, project.version -> 1.9.10, pom.groupId -> org.codehaus.jackson, groupId -> org.codehaus.jackson, project.groupId -> org.codehaus.jackson, version -> 1.9.10, pom.version -> 1.9.10, project.artifactId -> jackson-core-asl, pom.artifactId -> jackson-core-asl), None),/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/org/codehaus/jackson/jackson-core-asl/1.9.10/jackson-core-asl-1.9.10.jar)
[info]  docs:
[info]  optional:
[info]  compile:
[info]          com.codahale:jerkson_2.9.1:0.5.0:default (artifactId=jerkson_2.9.1, project.version=0.5.0, pom.groupId=com.codahale, groupId=com.codahale, project.groupId=com.codahale, version=0.5.0, pom.version=0.5.0, project.artifactId=jerkson_2.9.1, jackson.version=[1.9.0,2.0.0), scala.version=2.9.1, pom.artifactId=jerkson_2.9.1): (Artifact(jerkson_2.9.1, jar, jar, None, Vector(), Some(https://repo.typesafe.com/typesafe/maven-releases/com/codahale/jerkson_2.9.1/0.5.0/jerkson_2.9.1-0.5.0.jar), Map(artifactId -> jerkson_2.9.1, project.version -> 0.5.0, pom.groupId -> com.codahale, groupId -> com.codahale, project.groupId -> com.codahale, version -> 0.5.0, pom.version -> 0.5.0, project.artifactId -> jerkson_2.9.1, jackson.version -> [1.9.0,2.0.0), scala.version -> 2.9.1, pom.artifactId -> jerkson_2.9.1), None),/home/johannes/.coursier/cache/v1/https/repo.typesafe.com/typesafe/maven-releases/com/codahale/jerkson_2.9.1/0.5.0/jerkson_2.9.1-0.5.0.jar)
[info]          org.scala-lang:scala-library:2.9.1:default (artifactId=scala-library, project.version=2.9.1, pom.groupId=org.scala-lang, groupId=org.scala-lang, project.groupId=org.scala-lang, version=2.9.1, pom.version=2.9.1, project.artifactId=scala-library, pom.artifactId=scala-library): (Artifact(scala-library, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/scala-lang/scala-library/2.9.1/scala-library-2.9.1.jar), Map(artifactId -> scala-library, project.version -> 2.9.1, pom.groupId -> org.scala-lang, groupId -> org.scala-lang, project.groupId -> org.scala-lang, version -> 2.9.1, pom.version -> 2.9.1, project.artifactId -> scala-library, pom.artifactId -> scala-library), None),/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.9.1/scala-library-2.9.1.jar)
[info]          org.codehaus.jackson:jackson-mapper-asl:1.9.10:default (artifactId=jackson-mapper-asl, project.version=1.9.10, pom.groupId=org.codehaus.jackson, groupId=org.codehaus.jackson, project.groupId=org.codehaus.jackson, version=1.9.10, pom.version=1.9.10, project.artifactId=jackson-mapper-asl, pom.artifactId=jackson-mapper-asl): (Artifact(jackson-mapper-asl, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/codehaus/jackson/jackson-mapper-asl/1.9.10/jackson-mapper-asl-1.9.10.jar), Map(artifactId -> jackson-mapper-asl, project.version -> 1.9.10, pom.groupId -> org.codehaus.jackson, groupId -> org.codehaus.jackson, project.groupId -> org.codehaus.jackson, version -> 1.9.10, pom.version -> 1.9.10, project.artifactId -> jackson-mapper-asl, pom.artifactId -> jackson-mapper-asl), None),/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/org/codehaus/jackson/jackson-mapper-asl/1.9.10/jackson-mapper-asl-1.9.10.jar)
[info]          org.codehaus.jackson:jackson-core-asl:1.9.10:default (artifactId=jackson-core-asl, project.version=1.9.10, pom.groupId=org.codehaus.jackson, groupId=org.codehaus.jackson, project.groupId=org.codehaus.jackson, version=1.9.10, pom.version=1.9.10, project.artifactId=jackson-core-asl, pom.artifactId=jackson-core-asl): (Artifact(jackson-core-asl, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/codehaus/jackson/jackson-core-asl/1.9.10/jackson-core-asl-1.9.10.jar), Map(artifactId -> jackson-core-asl, project.version -> 1.9.10, pom.groupId -> org.codehaus.jackson, groupId -> org.codehaus.jackson, project.groupId -> org.codehaus.jackson, version -> 1.9.10, pom.version -> 1.9.10, project.artifactId -> jackson-core-asl, pom.artifactId -> jackson-core-asl), None),/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/org/codehaus/jackson/jackson-core-asl/1.9.10/jackson-core-asl-1.9.10.jar)
[info]  test-internal:
[info]          com.codahale:jerkson_2.9.1:0.5.0:default (artifactId=jerkson_2.9.1, project.version=0.5.0, pom.groupId=com.codahale, groupId=com.codahale, project.groupId=com.codahale, version=0.5.0, pom.version=0.5.0, project.artifactId=jerkson_2.9.1, jackson.version=[1.9.0,2.0.0), scala.version=2.9.1, pom.artifactId=jerkson_2.9.1): (Artifact(jerkson_2.9.1, jar, jar, None, Vector(), Some(https://repo.typesafe.com/typesafe/maven-releases/com/codahale/jerkson_2.9.1/0.5.0/jerkson_2.9.1-0.5.0.jar), Map(artifactId -> jerkson_2.9.1, project.version -> 0.5.0, pom.groupId -> com.codahale, groupId -> com.codahale, project.groupId -> com.codahale, version -> 0.5.0, pom.version -> 0.5.0, project.artifactId -> jerkson_2.9.1, jackson.version -> [1.9.0,2.0.0), scala.version -> 2.9.1, pom.artifactId -> jerkson_2.9.1), None),/home/johannes/.coursier/cache/v1/https/repo.typesafe.com/typesafe/maven-releases/com/codahale/jerkson_2.9.1/0.5.0/jerkson_2.9.1-0.5.0.jar)
[info]          org.scala-lang:scala-library:2.9.1:default (artifactId=scala-library, project.version=2.9.1, pom.groupId=org.scala-lang, groupId=org.scala-lang, project.groupId=org.scala-lang, version=2.9.1, pom.version=2.9.1, project.artifactId=scala-library, pom.artifactId=scala-library): (Artifact(scala-library, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/scala-lang/scala-library/2.9.1/scala-library-2.9.1.jar), Map(artifactId -> scala-library, project.version -> 2.9.1, pom.groupId -> org.scala-lang, groupId -> org.scala-lang, project.groupId -> org.scala-lang, version -> 2.9.1, pom.version -> 2.9.1, project.artifactId -> scala-library, pom.artifactId -> scala-library), None),/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.9.1/scala-library-2.9.1.jar)
[info]          org.codehaus.jackson:jackson-mapper-asl:1.9.10:default (artifactId=jackson-mapper-asl, project.version=1.9.10, pom.groupId=org.codehaus.jackson, groupId=org.codehaus.jackson, project.groupId=org.codehaus.jackson, version=1.9.10, pom.version=1.9.10, project.artifactId=jackson-mapper-asl, pom.artifactId=jackson-mapper-asl): (Artifact(jackson-mapper-asl, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/codehaus/jackson/jackson-mapper-asl/1.9.10/jackson-mapper-asl-1.9.10.jar), Map(artifactId -> jackson-mapper-asl, project.version -> 1.9.10, pom.groupId -> org.codehaus.jackson, groupId -> org.codehaus.jackson, project.groupId -> org.codehaus.jackson, version -> 1.9.10, pom.version -> 1.9.10, project.artifactId -> jackson-mapper-asl, pom.artifactId -> jackson-mapper-asl), None),/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/org/codehaus/jackson/jackson-mapper-asl/1.9.10/jackson-mapper-asl-1.9.10.jar)
[info]          org.codehaus.jackson:jackson-core-asl:1.9.10:default (artifactId=jackson-core-asl, project.version=1.9.10, pom.groupId=org.codehaus.jackson, groupId=org.codehaus.jackson, project.groupId=org.codehaus.jackson, version=1.9.10, pom.version=1.9.10, project.artifactId=jackson-core-asl, pom.artifactId=jackson-core-asl): (Artifact(jackson-core-asl, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/codehaus/jackson/jackson-core-asl/1.9.10/jackson-core-asl-1.9.10.jar), Map(artifactId -> jackson-core-asl, project.version -> 1.9.10, pom.groupId -> org.codehaus.jackson, groupId -> org.codehaus.jackson, project.groupId -> org.codehaus.jackson, version -> 1.9.10, pom.version -> 1.9.10, project.artifactId -> jackson-core-asl, pom.artifactId -> jackson-core-asl), None),/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/org/codehaus/jackson/jackson-core-asl/1.9.10/jackson-core-asl-1.9.10.jar)
[info]  scala-tool:
[info]          org.fusesource.jansi:jansi:1.4:compile (artifactId=jansi, project.build.sourceEncoding=UTF-8, project.version=1.4, pom.groupId=org.fusesource.jansi, groupId=org.fusesource.jansi, project.groupId=org.fusesource.jansi, version=1.4, release-altGitURL=scm:git:ssh://git@forge.fusesource.com/jansi.git, pom.version=1.4, project.artifactId=jansi, jansi-native-version=1.0, forge-project-id-uc=JANSI, forge-project-id=jansi, pom.artifactId=jansi): (Artifact(jansi, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/fusesource/jansi/jansi/1.4/jansi-1.4.jar), Map(artifactId -> jansi, project.build.sourceEncoding -> UTF-8, project.version -> 1.4, pom.groupId -> org.fusesource.jansi, groupId -> org.fusesource.jansi, project.groupId -> org.fusesource.jansi, version -> 1.4, release-altGitURL -> scm:git:ssh://git@forge.fusesource.com/jansi.git, pom.version -> 1.4, project.artifactId -> jansi, jansi-native-version -> 1.0, forge-project-id-uc -> JANSI, forge-project-id -> jansi, pom.artifactId -> jansi), None),/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/org/fusesource/jansi/jansi/1.4/jansi-1.4.jar)
[info]          org.scala-lang:scala-compiler:2.9.1:optional (artifactId=scala-compiler, project.version=2.9.1, pom.groupId=org.scala-lang, groupId=org.scala-lang, project.groupId=org.scala-lang, version=2.9.1, pom.version=2.9.1, project.artifactId=scala-compiler, pom.artifactId=scala-compiler): (Artifact(scala-compiler, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/scala-lang/scala-compiler/2.9.1/scala-compiler-2.9.1.jar), Map(artifactId -> scala-compiler, project.version -> 2.9.1, pom.groupId -> org.scala-lang, groupId -> org.scala-lang, project.groupId -> org.scala-lang, version -> 2.9.1, pom.version -> 2.9.1, project.artifactId -> scala-compiler, pom.artifactId -> scala-compiler), None),/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-compiler/2.9.1/scala-compiler-2.9.1.jar)
[info]          org.scala-lang:jline:2.9.1:compile (artifactId=jline, project.version=2.9.1, pom.groupId=org.scala-lang, groupId=org.scala-lang, project.groupId=org.scala-lang, version=2.9.1, pom.version=2.9.1, project.artifactId=jline, pom.artifactId=jline): (Artifact(jline, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/scala-lang/jline/2.9.1/jline-2.9.1.jar), Map(artifactId -> jline, project.version -> 2.9.1, pom.groupId -> org.scala-lang, groupId -> org.scala-lang, project.groupId -> org.scala-lang, version -> 2.9.1, pom.version -> 2.9.1, project.artifactId -> jline, pom.artifactId -> jline), None),/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/org/scala-lang/jline/2.9.1/jline-2.9.1.jar)
[info]          org.scala-lang:scala-library:2.9.1:optional (artifactId=scala-library, project.version=2.9.1, pom.groupId=org.scala-lang, groupId=org.scala-lang, project.groupId=org.scala-lang, version=2.9.1, pom.version=2.9.1, project.artifactId=scala-library, pom.artifactId=scala-library): (Artifact(scala-library, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/scala-lang/scala-library/2.9.1/scala-library-2.9.1.jar), Map(artifactId -> scala-library, project.version -> 2.9.1, pom.groupId -> org.scala-lang, groupId -> org.scala-lang, project.groupId -> org.scala-lang, version -> 2.9.1, pom.version -> 2.9.1, project.artifactId -> scala-library, pom.artifactId -> scala-library), None),/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.9.1/scala-library-2.9.1.jar)
[info]  sources:
[info]  runtime:
[info]          com.codahale:jerkson_2.9.1:0.5.0:default (artifactId=jerkson_2.9.1, project.version=0.5.0, pom.groupId=com.codahale, groupId=com.codahale, project.groupId=com.codahale, version=0.5.0, pom.version=0.5.0, project.artifactId=jerkson_2.9.1, jackson.version=[1.9.0,2.0.0), scala.version=2.9.1, pom.artifactId=jerkson_2.9.1): (Artifact(jerkson_2.9.1, jar, jar, None, Vector(), Some(https://repo.typesafe.com/typesafe/maven-releases/com/codahale/jerkson_2.9.1/0.5.0/jerkson_2.9.1-0.5.0.jar), Map(artifactId -> jerkson_2.9.1, project.version -> 0.5.0, pom.groupId -> com.codahale, groupId -> com.codahale, project.groupId -> com.codahale, version -> 0.5.0, pom.version -> 0.5.0, project.artifactId -> jerkson_2.9.1, jackson.version -> [1.9.0,2.0.0), scala.version -> 2.9.1, pom.artifactId -> jerkson_2.9.1), None),/home/johannes/.coursier/cache/v1/https/repo.typesafe.com/typesafe/maven-releases/com/codahale/jerkson_2.9.1/0.5.0/jerkson_2.9.1-0.5.0.jar)
[info]          org.scala-lang:scala-library:2.9.1:default (artifactId=scala-library, project.version=2.9.1, pom.groupId=org.scala-lang, groupId=org.scala-lang, project.groupId=org.scala-lang, version=2.9.1, pom.version=2.9.1, project.artifactId=scala-library, pom.artifactId=scala-library): (Artifact(scala-library, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/scala-lang/scala-library/2.9.1/scala-library-2.9.1.jar), Map(artifactId -> scala-library, project.version -> 2.9.1, pom.groupId -> org.scala-lang, groupId -> org.scala-lang, project.groupId -> org.scala-lang, version -> 2.9.1, pom.version -> 2.9.1, project.artifactId -> scala-library, pom.artifactId -> scala-library), None),/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.9.1/scala-library-2.9.1.jar)
[info]          org.codehaus.jackson:jackson-mapper-asl:1.9.10:default (artifactId=jackson-mapper-asl, project.version=1.9.10, pom.groupId=org.codehaus.jackson, groupId=org.codehaus.jackson, project.groupId=org.codehaus.jackson, version=1.9.10, pom.version=1.9.10, project.artifactId=jackson-mapper-asl, pom.artifactId=jackson-mapper-asl): (Artifact(jackson-mapper-asl, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/codehaus/jackson/jackson-mapper-asl/1.9.10/jackson-mapper-asl-1.9.10.jar), Map(artifactId -> jackson-mapper-asl, project.version -> 1.9.10, pom.groupId -> org.codehaus.jackson, groupId -> org.codehaus.jackson, project.groupId -> org.codehaus.jackson, version -> 1.9.10, pom.version -> 1.9.10, project.artifactId -> jackson-mapper-asl, pom.artifactId -> jackson-mapper-asl), None),/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/org/codehaus/jackson/jackson-mapper-asl/1.9.10/jackson-mapper-asl-1.9.10.jar)
[info]          org.codehaus.jackson:jackson-core-asl:1.9.10:default (artifactId=jackson-core-asl, project.version=1.9.10, pom.groupId=org.codehaus.jackson, groupId=org.codehaus.jackson, project.groupId=org.codehaus.jackson, version=1.9.10, pom.version=1.9.10, project.artifactId=jackson-core-asl, pom.artifactId=jackson-core-asl): (Artifact(jackson-core-asl, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/codehaus/jackson/jackson-core-asl/1.9.10/jackson-core-asl-1.9.10.jar), Map(artifactId -> jackson-core-asl, project.version -> 1.9.10, pom.groupId -> org.codehaus.jackson, groupId -> org.codehaus.jackson, project.groupId -> org.codehaus.jackson, version -> 1.9.10, pom.version -> 1.9.10, project.artifactId -> jackson-core-asl, pom.artifactId -> jackson-core-asl), None),/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/org/codehaus/jackson/jackson-core-asl/1.9.10/jackson-core-asl-1.9.10.jar)
[info]  runtime-internal:
[info]          com.codahale:jerkson_2.9.1:0.5.0:default (artifactId=jerkson_2.9.1, project.version=0.5.0, pom.groupId=com.codahale, groupId=com.codahale, project.groupId=com.codahale, version=0.5.0, pom.version=0.5.0, project.artifactId=jerkson_2.9.1, jackson.version=[1.9.0,2.0.0), scala.version=2.9.1, pom.artifactId=jerkson_2.9.1): (Artifact(jerkson_2.9.1, jar, jar, None, Vector(), Some(https://repo.typesafe.com/typesafe/maven-releases/com/codahale/jerkson_2.9.1/0.5.0/jerkson_2.9.1-0.5.0.jar), Map(artifactId -> jerkson_2.9.1, project.version -> 0.5.0, pom.groupId -> com.codahale, groupId -> com.codahale, project.groupId -> com.codahale, version -> 0.5.0, pom.version -> 0.5.0, project.artifactId -> jerkson_2.9.1, jackson.version -> [1.9.0,2.0.0), scala.version -> 2.9.1, pom.artifactId -> jerkson_2.9.1), None),/home/johannes/.coursier/cache/v1/https/repo.typesafe.com/typesafe/maven-releases/com/codahale/jerkson_2.9.1/0.5.0/jerkson_2.9.1-0.5.0.jar)
[info]          org.scala-lang:scala-library:2.9.1:default (artifactId=scala-library, project.version=2.9.1, pom.groupId=org.scala-lang, groupId=org.scala-lang, project.groupId=org.scala-lang, version=2.9.1, pom.version=2.9.1, project.artifactId=scala-library, pom.artifactId=scala-library): (Artifact(scala-library, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/scala-lang/scala-library/2.9.1/scala-library-2.9.1.jar), Map(artifactId -> scala-library, project.version -> 2.9.1, pom.groupId -> org.scala-lang, groupId -> org.scala-lang, project.groupId -> org.scala-lang, version -> 2.9.1, pom.version -> 2.9.1, project.artifactId -> scala-library, pom.artifactId -> scala-library), None),/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.9.1/scala-library-2.9.1.jar)
[info]          org.codehaus.jackson:jackson-mapper-asl:1.9.10:default (artifactId=jackson-mapper-asl, project.version=1.9.10, pom.groupId=org.codehaus.jackson, groupId=org.codehaus.jackson, project.groupId=org.codehaus.jackson, version=1.9.10, pom.version=1.9.10, project.artifactId=jackson-mapper-asl, pom.artifactId=jackson-mapper-asl): (Artifact(jackson-mapper-asl, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/codehaus/jackson/jackson-mapper-asl/1.9.10/jackson-mapper-asl-1.9.10.jar), Map(artifactId -> jackson-mapper-asl, project.version -> 1.9.10, pom.groupId -> org.codehaus.jackson, groupId -> org.codehaus.jackson, project.groupId -> org.codehaus.jackson, version -> 1.9.10, pom.version -> 1.9.10, project.artifactId -> jackson-mapper-asl, pom.artifactId -> jackson-mapper-asl), None),/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/org/codehaus/jackson/jackson-mapper-asl/1.9.10/jackson-mapper-asl-1.9.10.jar)
[info]          org.codehaus.jackson:jackson-core-asl:1.9.10:default (artifactId=jackson-core-asl, project.version=1.9.10, pom.groupId=org.codehaus.jackson, groupId=org.codehaus.jackson, project.groupId=org.codehaus.jackson, version=1.9.10, pom.version=1.9.10, project.artifactId=jackson-core-asl, pom.artifactId=jackson-core-asl): (Artifact(jackson-core-asl, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/codehaus/jackson/jackson-core-asl/1.9.10/jackson-core-asl-1.9.10.jar), Map(artifactId -> jackson-core-asl, project.version -> 1.9.10, pom.groupId -> org.codehaus.jackson, groupId -> org.codehaus.jackson, project.groupId -> org.codehaus.jackson, version -> 1.9.10, pom.version -> 1.9.10, project.artifactId -> jackson-core-asl, pom.artifactId -> jackson-core-asl), None),/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/org/codehaus/jackson/jackson-core-asl/1.9.10/jackson-core-asl-1.9.10.jar)
```

Output of `show externalDependencyClasspath` in sbt 1.2.8:

```
[info] * Attributed(/home/johannes/.ivy2/cache/org.scala-lang/scala-library/jars/scala-library-2.9.1.jar)
[info] * Attributed(/home/johannes/.ivy2/cache/com.codahale/jerkson_2.9.1/jars/jerkson_2.9.1-0.5.0.jar)
[info] * Attributed(/home/johannes/.ivy2/cache/org.codehaus.jackson/jackson-core-asl/jars/jackson-core-asl-1.9.11.jar)
[info] * Attributed(/home/johannes/.ivy2/cache/org.codehaus.jackson/jackson-mapper-asl/jars/jackson-mapper-asl-1.9.11.jar)
```

Output of `show externalDependencyClasspath` in sbt 1.3.0-RC1 (note that the wrong version 1.9.10 of jackson is chosen):

```
[info] * Attributed(/home/johannes/.coursier/cache/v1/https/repo.typesafe.com/typesafe/maven-releases/com/codahale/jerkson_2.9.1/0.5.0/jerkson_2.9.1-0.5.0.jar)
[info] * Attributed(/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.9.1/scala-library-2.9.1.jar)
[info] * Attributed(/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/org/codehaus/jackson/jackson-mapper-asl/1.9.10/jackson-mapper-asl-1.9.10.jar)
[info] * Attributed(/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/org/codehaus/jackson/jackson-core-asl/1.9.10/jackson-core-asl-1.9.10.jar)
```

**Comment by jrudolph** [comment]
> Another big problem is that eviction is now working differently than before. How was coursier support tested if that most basic of features has not been tested for regressions?

See the update in #4710 on that issue. It seems that particular artifact combination leads to a weird edge case, where ivy seemed to have pulled version 1.9.11 of jackson out of nowhere and the coursier behavior is quite defensible.

**Comment by jrudolph** [comment]
So the remaining problem with eviction is that it is not reported any more it seems (also `updateFull` doesn't show evicted versions at all):

```
version := "0.1.0-SNAPSHOT"

scalaVersion := "2.12.8"

libraryDependencies ++= Seq(
  "com.typesafe.akka" %% "akka-http" % "10.1.8",
  "com.typesafe.akka" %% "akka-http-core" % "10.1.7" // as another version of http-core
)
```

1.2.8:

```
sbt:test> show externalDependencyClasspath
[info] * Attributed(/tmp/test/ivy/cache/org.scala-lang/scala-library/jars/scala-library-2.12.8.jar)
[info] * Attributed(/tmp/test/ivy/cache/com.typesafe.akka/akka-http_2.12/jars/akka-http_2.12-10.1.8.jar)
[info] * Attributed(/tmp/test/ivy/cache/com.typesafe.akka/akka-http-core_2.12/jars/akka-http-core_2.12-10.1.8.jar)
[info] * Attributed(/tmp/test/ivy/cache/com.typesafe.akka/akka-parsing_2.12/jars/akka-parsing_2.12-10.1.8.jar)
[success] Total time: 5 s, completed May 20, 2019 2:21:23 PM
sbt:test> evicted
[info] Updating ...
[info] Done updating.
[info] Here are other dependency conflicts that were resolved:
[info] 	* com.typesafe.akka:akka-http-core_2.12:10.1.8 is selected over 10.1.7
[info] 	    +- com.typesafe.akka:akka-http_2.12:10.1.8 ()         (depends on 10.1.8)
[info] 	    +- default:test_2.12:0.1.0-SNAPSHOT                   (depends on 10.1.7)
```

1.3.0-RC1:

```
sbt:test> show externalDependencyClasspath
[info] * Attributed(/home/johannes/.sbt/boot/scala-2.12.8/lib/scala-library.jar)
[info] * Attributed(/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/com/typesafe/akka/akka-parsing_2.12/10.1.8/akka-parsing_2.12-10.1.8.jar)
[info] * Attributed(/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/com/typesafe/akka/akka-http-core_2.12/10.1.8/akka-http-core_2.12-10.1.8.jar)
[info] * Attributed(/home/johannes/.coursier/cache/v1/https/repo1.maven.org/maven2/com/typesafe/akka/akka-http_2.12/10.1.8/akka-http_2.12-10.1.8.jar)
[success] Total time: 1 s, completed May 20, 2019 2:21:56 PM
sbt:test> evicted
[success] Total time: 0 s, completed May 20, 2019 2:22:01 PM
```

**Comment by eatkins** [comment]
@eed3si9n do we need to in-source the sbt-coursier changes in https://github.com/coursier/sbt-coursier/pull/87 to resolve this issue in sbt using the built in coursier?

**Comment by eed3si9n** [comment]
I am guessing that we need a new version lm-coursier published.

**Comment by jrudolph** [comment]
This is a slight bit better in 1.3.0-RC3 but still doesn't work as before. It seems caller information for the top-level dependencies is missing.

For

```
scalaVersion := "2.9.2"

libraryDependencies ++= Seq(
  "org.slf4j" % "slf4j-api" % "1.7.2",
  "ch.qos.logback" % "logback-classic" % "1.0.7"
)
```

<details>
<summary>Update report from 1.2.8</summary>

```
[info] 	ch.qos.logback:logback-classic
[info] 		- 1.0.7
[info] 			status: release
[info] 			publicationDate: 2012-08-24T00:13:47+02:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			homepage: http://logback.qos.ch
[info] 			isDefault: false
[info] 			configurations: default, compile, runtime, default(compile), master
[info] 			licenses: (Eclipse Public License - v 1.0,Some(http://www.eclipse.org/legal/epl-v10.html)), (GNU Lesser General Public License,Some(http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html))
[info] 			callers: default:ignorescalalibrary_2.9.2:0.1.0-SNAPSHOT
[info] 
[info] 	ch.qos.logback:logback-core
[info] 		- 1.0.7
[info] 			status: release
[info] 			publicationDate: 2012-08-24T00:13:02+02:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			homepage: http://logback.qos.ch
[info] 			isDefault: false
[info] 			configurations: master(*), compile, runtime(*), runtime, compile(*), master
[info] 			licenses: (Eclipse Public License - v 1.0,Some(http://www.eclipse.org/legal/epl-v10.html)), (GNU Lesser General Public License,Some(http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html))
[info] 			callers: ch.qos.logback:logback-classic:1.0.7
[info] 
[info] 	org.slf4j:slf4j-api
[info] 		- 1.7.2
[info] 			status: release
[info] 			publicationDate: 2012-10-11T20:55:24+02:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			homepage: http://www.slf4j.org
[info] 			isDefault: false
[info] 			configurations: default, compile, runtime, default(compile), master, master(*), runtime(*), compile(*)
[info] 			callers: default:ignorescalalibrary_2.9.2:0.1.0-SNAPSHOT, ch.qos.logback:logback-classic:1.0.7
[info] 		- 1.6.6
[info] 			evicted: true
[info] 			evictedData: latest-revision
[info] 			configurations: master(*), compile, runtime(*), runtime, compile(*), master
[info] 			callers: ch.qos.logback:logback-classic:1.0.7
[info] 
[info] 	org.scala-lang:scala-library
[info] 		- 2.9.2
[info] 			status: release
[info] 			publicationDate: 2012-04-13T14:29:21+02:00
[info] 			resolver: sbt-chain
[info] 			artifactResolver: sbt-chain
[info] 			evicted: false
[info] 			homepage: http://www.scala-lang.org/
[info] 			isDefault: false
[info] 			configurations: default, compile, runtime, default(compile), master
[info] 			licenses: (BSD-like,Some(http://www.scala-lang.org/downloads/license.html))
[info] 			callers: default:ignorescalalibrary_2.9.2:0.1.0-SNAPSHOT
```

</details>

<details>
<summary>Update report from 1.3.0-RC3</summary>

```
[info] 	org.slf4j:slf4j-api
[info] 		- 1.7.2
[info] 			evicted: false
[info] 			homepage: http://www.slf4j.org
[info] 			configurations: test, optional, compile, default, runtime
[info] 			callers: ch.qos.logback:logback-classic:1.0.7
[info] 		- 1.6.6
[info] 			evicted: true
[info] 			evictedData: version selection
[info] 			homepage: http://www.slf4j.org
[info] 			configurations: test, optional, compile, default, runtime
[info] 			callers: ch.qos.logback:logback-classic:1.0.7
[info] 
[info] 	ch.qos.logback:logback-core
[info] 		- 1.0.7
[info] 			evicted: false
[info] 			homepage: http://logback.qos.ch
[info] 			configurations: test, optional, compile, default, runtime
[info] 			callers: ch.qos.logback:logback-classic:1.0.7
[info] 
[info] 	org.scala-lang:scala-library
[info] 		- 2.9.2
[info] 			evicted: false
[info] 			homepage: http://www.scala-lang.org/
[info] 			configurations: test, optional, compile, default, runtime
[info] 
[info] 	ch.qos.logback:logback-classic
[info] 		- 1.0.7
[info] 			evicted: false
[info] 			homepage: http://logback.qos.ch
[info] 			configurations: test, optional, compile, default, runtime
```

</details>

**Comment by bjaglin** [comment]
> This is a slight bit better in 1.3.0-RC3 but still doesn't work as before. It seems caller information for the top-level dependencies is missing.

Still the same behavior in 1.3.0-RC5: despite `includeCallers=true`, caller information is missing for top-level dependencies.

**Comment by jrudolph** [comment]
What's the state here? sbt 1.3.0 has been released but this regression has not been fixed. How can we make sure it will be fixed quickly? Until then, sbt-dependency-graph will be broken unless you disable coursier. E.g. it's one of the remaining headaches in updating the akka build to 1.3.0.

**Comment by eed3si9n** [comment]
@jrudolph This has slipped through our attention. We'll look into it for 1.3.1.

**Comment by jrudolph** [comment]
Thanks, Eugene!

**Comment by eed3si9n** [comment]
I think this was fixed in upstream (https://github.com/coursier/sbt-coursier/pull/87).

```
sbt:foo> dependencyTree
[info] Updating ...
[info] Done updating.
[info] com.example:foo_2.13:0.1.0-SNAPSHOT [S]
[info]   +-com.typesafe:config:1.3.4
[info]
[success] Total time: 0 s, completed Sep 17, 2019 5:29:31 PM
```

`dependencyTree` seems to work using `addSbtPlugin("net.virtual-void" % "sbt-dependency-graph" % "0.10.0-RC1")`.

**Comment by eed3si9n** [comment]
> Still the same behavior in 1.3.0-RC5: despite `includeCallers=true`, caller information is missing for top-level dependencies.

ok. Testing this using akka-stream I get something this:

```
[info] com.typesafe.akka:akka-stream_2.12:2.5-SNAPSHOT
[info] com.typesafe:ssl-config-core_2.12:0.3.7 [S]
[info]   +-com.typesafe:config:1.3.3
[info]   +-org.scala-lang.modules:scala-parser-combinators_2.12:1.1.1 [S]
[info]
[info] org.reactivestreams:reactive-streams:1.0.2
[success] Total time: 1 s, completed Sep 17, 2019 5:44:30 PM
```

**Comment by eed3si9n** [comment]
caller info is missing for `org.reactivestreams:reactive-streams` and `com.typesafe:ssl-config-core_2.12`

```
[info] 	compile:
[info] 	com.typesafe:config
[info] 		- 1.3.3
[info] 			evicted: false
[info] 			homepage: https://github.com/lightbend/config
[info] 			configurations: test, optional, compile, default, runtime
[info] 			callers: com.typesafe:ssl-config-core_2.12:0.3.7, com.typesafe.akka:akka-actor_2.12:2.5-SNAPSHOT
[info] 
[info] 	org.scala-lang:scala-library
[info] 		- 2.12.8
[info] 			evicted: false
[info] 			homepage: http://www.scala-lang.org/
[info] 			extraAttributes: Map(info.apiURL -> http://www.scala-lang.org/api/2.12.8/)
[info] 			configurations: test, optional, compile, default, runtime
[info] 			callers: com.typesafe.akka:akka-protobuf_2.12:2.5-SNAPSHOT, org.scala-lang.modules:scala-parser-combinators_2.12:1.1.1, com.typesafe:ssl-config-core_2.12:0.3.7, org.scala-lang.modules:scala-java8-compat_2.12:0.8.0, com.typesafe.akka:akka-actor_2.12:2.5-SNAPSHOT
[info] 
[info] 	org.scala-lang.modules:scala-parser-combinators_2.12
[info] 		- 1.1.1
[info] 			evicted: false
[info] 			homepage: http://www.scala-lang.org/
[info] 			configurations: test, optional, compile, default, runtime
[info] 			callers: com.typesafe:ssl-config-core_2.12:0.3.7
[info] 
[info] 	org.reactivestreams:reactive-streams
[info] 		- 1.0.2
[info] 			evicted: false
[info] 			homepage: http://www.reactive-streams.org/
[info] 			configurations: test, optional, compile, default, runtime
[info] 
[info] 	org.scala-lang.modules:scala-java8-compat_2.12
[info] 		- 0.8.0
[info] 			evicted: false
[info] 			homepage: http://www.scala-lang.org/
[info] 			configurations: test, optional, compile, default, runtime
[info] 			callers: com.typesafe.akka:akka-actor_2.12:2.5-SNAPSHOT
[info] 
[info] 	com.typesafe:ssl-config-core_2.12
[info] 		- 0.3.7
[info] 			evicted: false
[info] 			homepage: https://github.com/lightbend/ssl-config
[info] 			configurations: test, optional, compile, default, runtime
[info] 
```

**Comment by eed3si9n** [comment]
I've opened an issue upstream - https://github.com/coursier/coursier/issues/1375

**Comment by jrudolph** [comment]
What's the state here? People are asking me what is going on with sbt-dependency-graph and there's no satisfactory answer I could give.

To be clear, I think this is a QA issue in sbt in the first place. This regression has been known for months and the sbt 1.3.0 (and 1.3.1, 1.3.2, 1.3.3) release still went ahead changing the default to coursier in sbt 1.3.0. Now saying this is "blocked on upstream" does not really help. By making coursier the default, sbt is now responsible for making it work correctly.

E.g. to demonstrate the difficulties: in the akka-http build we depend on sbt-dependency-graph for some analytics so until now we couldn't update to sbt 1.3.0. But that also means that we cannot update mima because it requires sbt 1.3.x, etc. We will probably just disable coursier for akka-http but it would be better if we wouldn't have to.

**Comment by eed3si9n** [comment]
> To be clear, I think this is a QA issue in sbt in the first place. This regression has been known for months and the sbt 1.3.0 (and 1.3.1, 1.3.2, 1.3.3) release still went ahead changing the default to coursier in sbt 1.3.0. Now saying this is "blocked on upstream" does not really help. By making coursier the default, sbt is now responsible for making it work correctly.

I guess the overall issue is that the output of dependency graph is not part of our integration test. Both sbt and Coursier projects have tried to fix this in good faith (see https://github.com/coursier/sbt-coursier/pull/87) but ultimately the fix didn't fully fix the issue.

"Blocked on Upstream" isn't a signal that I am abdicating the responsibility, but it's a label that I use to signal that the source change must take place elsewhere. In some situations, I've waded into Coursier code base and patched it myself.

**Comment by dwijnand** [comment]
cc @alexarchambault, do you think you'll have some spare bandwidth to try and reverse this regression?  I'm one of those people who recently asked Johannes how to investigate a dependency issue in an sbt 1.3 build.

**Comment by eed3si9n** [comment]
Here's my PR to sbt-coursier - https://github.com/coursier/sbt-coursier/pull/156

**Comment by albuch** [comment]
Hi @eed3si9n,
in which release will this fix be available? I couldn't find the issues #4706 nor #5240 mentioned in any release notes yet.

**Comment by eed3si9n** [comment]
@albuch see https://github.com/sbt/sbt/releases/tag/v1.3.4
